docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,854 @@
+  }
+}
+```
+
+*spawning validation work*
+```rust
+fn spawn_validation_work(candidate, parachain head, validation function) {
+  asynchronously {
+    let pov = (fetch pov block).await
+
+    // dispatched to sub-process (OS process) pool.
+    let valid = validate_candidate(candidate, validation function, parachain head, pov).await;
+    if valid {
+      // make PoV available for later distribution.
+      // sign and dispatch `valid` statement to network if we have not seconded the given candidate.
+    } else {
+      // sign and dispatch `invalid` statement to network.
+    }
+  }
+}
+```
+
+*fetch pov block*
+
+Create a `(sender, receiver)` pair.
+Dispatch a `PovFetchProcessMessage(relay_parent, candidate_hash, sender)` and listen on the receiver for a response.
+
+*on receiving CandidateBackingProcessMessage*
+* If the message is a `CandidateBackingProcessMessage::RegisterBackingWatcher`, register the watcher and trigger it each time a new candidate is backed. Also trigger it once initially if there are any backed candidates at the time of receipt.
+* If the message is a `CandidateBackingProcessMessage::Second`, sign and dispatch a `Seconded` statement only if we have not seconded any other candidate and have not signed a `Valid` statement for the requested candidate. Signing both a `Seconded` and `Valid` message is a double-voting misbehavior with a heavy penalty, and this could occur if another validator has seconded the same candidate and we've received their message before the internal seconding request.
+
+(TODO: send statements to Statement Distribution Process, handle shutdown signal from parent process)
+
+---
+
+### Candidate Proposal Process
+
+[TODO: get candidate from collator, feed to candidate backing process. ]
+
+### Secondary Checking
+
+[TODO]
+
+### Validator Assignment Process
+
+[TODO]
+
+### Collation Distribution Process
+
+[TODO]
+
+### Availability Distribution Process
+
+[TODO]
+
+----
+
+## Data Structures and Types
+
+[TODO]
+* CandidateReceipt
+* CandidateCommitments
+* AbridgedCandidateReceipt
+* GlobalValidationSchedule
+* LocalValidationData
+
+#### Block Import Event
+```rust
+/// Indicates that a new block has been added to the blockchain.
+struct BlockImportEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header itself.
+  header: Header,
+  /// Whether this block is considered the head of the best chain according to the 
+  /// event emitter's fork-choice rule.
+  new_best: bool,
+}
+```
+
+#### Block Finalization Event
+```rust
+/// Indicates that a new block has been finalized.
+struct BlockFinalizationEvent {
+  /// The block header-hash.
+  hash: Hash,
+  /// The header of the finalized block.
+  header: Header,
+}
+```
+
+#### Statement Type
+```rust
+/// A statement about the validity of a parachain candidate.
+enum Statement {
+  /// A statement about a new candidate being seconded by a validator. This is an implicit validity vote.
+  Seconded(CandidateReceipt),
```

Comment:
```suggestion
  Seconded(Hash),
```

Per the first line of the  "Signed Statement Type" description.

Reply:
No, we actually want the `CandidateReceipt` to be included in full here. It's just that the payload that's actually signed in the `SignedStatement` shouldn't encode the full candidate receipt, but only its hash. That allows us to omit the receipt elsewhere. Do you have any ideas on how to communicate that better?

The reason we want to include the `CandidateReceipt` in the `Statement::Seconded` is so we can ensure we have all relevant `CandidateReceipt`s in a data structure which only stores `Statement`s. Otherwise you'd always have to carry along an `Option<CandidateReceipt>` which would only be `Some` if the `Statement` was `Seconded`, as seconding is the introductory point of a candidate receipt to the import pipeline.

Reply:
These parachain validator groups are very small so repeating the `CandidateReceipt`s works fine.  It's their collection into one backed extrinsic that deduplicates this data.